zentral/contrib/osquery/serializers.py

```diff
@@ -152,3 +153,46 @@ def iter_query_defaults(self, pack_slug):
                 "compliance_check": query_data.get("compliance_check") or False
             }
             yield query_slug, pack_query_defaults, query_defaults
+
+
+class QuerySerializer(serializers.ModelSerializer):
+    compliance_check_enabled = serializers.BooleanField(default=False)
+
+    class Meta:
+        model = Query
+        exclude = ("compliance_check",)
+
+    def validate(self, data):
+        if data.get("compliance_check_enabled"):
+            sql = data.get("sql")
+            if sql is not None and 'ztl_status' not in sql:
+                raise serializers.ValidationError('{compliance_check_enabled: true} '
+                                                  'only if query contains ztl_status')
+            try:
+                pack_query = data.instance.packquery
+            except AttributeError:
+                pass
+            else:
+                if not pack_query.snapshot_mode:
+                    raise serializers.ValidationError('{compliance_check_enabled: true} '
+                                                      'This query is scheduled in diff mode in '
+                                                      f'the {pack_query.pack} pack')
+        return data
+
+    def create(self, validated_data):
+        compliance_check_enabled = validated_data.pop("compliance_check_enabled")
+        query = super().create(validated_data)
+        sync_query_compliance_check(query, compliance_check_enabled)
+        return query
+
+    def update(self, instance, validated_data):
+        compliance_check_enabled = validated_data.pop("compliance_check_enabled")
+        if instance.pk:
+            sql = validated_data.get("sql")
+            if sql and sql != instance.sql:
+                instance.version += 1
```

Comment:
Maybe try to use F("version") + 1, and see if it works? This way we are sure the version is incremented from the current value in the DB.